lib/src/query_service.dart

```diff
@@ -135,4 +108,47 @@ class QueryService {
         return RequestArgs();
     }
   }
+
+  /// Generates query based on user inputs.
+  static RequestArgs _generateReportingQuery(
+      QueryType queryType,
+      String advertiserId,
+      String insertionOrderId,
+      DateTime startDate,
+      DateTime endDate) {
+    final parameter = ReportingQueryParameter()
+      ..metadata = (ReportingQueryParameter_Metadata()
+        ..title = '"DV360-excel-plugin-query"'
+        ..dataRange = '"CUSTOM_DATES"'
+        ..format = '"EXCEL_CSV"')
+      ..params = (ReportingQueryParameter_Params()
+        ..metrics.add('"METRIC_REVENUE_USD"')
+        ..metrics.add('"METRIC_IMPRESSIONS"')
+        ..groupBys.add('"FILTER_INSERTION_ORDER"')
+        ..groupBys.add('"FILTER_DATE"')
+        ..filters.add(ReportingQueryParameter_Params_Filters()
+          ..type = '"FILTER_ADVERTISER"'
+          ..value = advertiserId))
+      ..reportDataStartTimeMs = Int64(startDate.millisecondsSinceEpoch)
+      ..reportDataEndTimeMs = Int64(endDate.millisecondsSinceEpoch);
+
+    switch (queryType) {
+      case QueryType.byAdvertiser:
+        break;
+
```

Comment:
is there an extra space here?

Reply:
Line 138? Is the line between two cases optional?